src/main/java/io/cryostat/net/web/http/api/v2/graph/GraphQLPostHandler.java

```diff
@@ -107,6 +114,78 @@ public void handle(RoutingContext ctx) {
                     .get()) {
                 throw new ApiException(401);
             }
+
+            String targetId = ctx.pathParam("targetId");
```

Comment:
I'm not sure but how is the GraphQL handler able to use path params in the requests to `/api/v2.2/graphql`? Especially if we querying on multiple targets at once?

I've been trying GraphQL requests but they don't seem to work either, e.g.
```bash
http --auth-type=basic --auth=user:pass :8181/api/v2.2/graphql query=@my-query.graphql X-JMX-Authorization:"Basic $(echo -n admin:adminpass123 | base64)"
```
and `my-query.graphql`:
```graphql
query { 
    targetNodes(filter: { annotations: "PORT == 9094" }) {
        doStartRecording(recording: { 
            name: "graphql-itest", 
            duration: 30,
            template: "Profiling", 
            templateType: "TARGET", 
            metadata: {
                labels: {
                    newLabel: someValue 
                } 
            } 
        })
        { 
            name state duration 
        }
    } 
}
```
results in 
```bash
...
...
"message": "Exception while fetching data (/targetNodes[0]/doStartRecording) : org.openjdk.jmc.rjmx.ConnectionException caused by java.lang.SecurityException: Authentication failed! Invalid username or password",
...
...
```

Reply:
Ah good catch. Bad case of copy-paste brain fart. I'll fix it up tomorrow.

Reply:
I'm actually not sure exactly what I want to do here. This code needs to move into something like the `TargetNodeFetcher`/`TargetNodesRecurseFetcher` I think, but a single request can actually be for multiple different targets. The web-client frontend doesn't have any UI that would allow the user to supply multiple credentials in case the targets may need different credentials, but conceptually that's possible, and the resulting map could be encoded in headers like `X-JMX-Authorization targetId1=encodedCredentials1,targetId2=encodedCredentials2`.

Maybe that's too much and, at least for now, the assumption should just be that if session credentials are passed with a request then they should be valid for *all* targets affected by the request execution.

----

https://github.com/cryostatio/cryostat/blob/73f42066c48fcd4f2cc173aaec15f9a1fec258cc/src/main/java/io/cryostat/net/web/http/api/v2/graph/GraphModule.java#L124

Each stage of the fetcher/mutator pipeline actually runs async and so may be executed by a different thread like I described above, so the `ThreadLocal` won't work properly here if it's set early in the call stack like by the `GraphQLPostHandler`. But, each fetcher/mutator stage can do `        GraphQLHandler.getRoutingContext(environment.getGraphQlContext()).request().getHeader("X-JMX-Authorization");
` to directly check for session credentials, and can set the `ThreadLocal` at that point. This should work equivalently well (or not) as the rest of the solution is working for the HTTP API abstract classes.